pi-hole/rootfs/etc/unbound/unbound.conf

```diff
@@ -0,0 +1,52 @@
+# This config was borrowed from: https://docs.pi-hole.net/guides/unbound/
+server:
+    # If no logfile is specified, syslog is used
+    # logfile: "/var/log/unbound/unbound.log"
+    verbosity: 0
+
+    port: 5353
+    do-ip4: yes
+    do-udp: yes
+    do-tcp: yes
+
+    # May be set to yes if you have IPv6 connectivity
+    do-ip6: no
```

Comment:
This is a good example of a config option that should not be hardcoded into the dockerbuild.
It would be prefered to keep the whole unbound.conf outside the docker, there are many many different use cases for an unbound server.

Reply:
I can expose more options to the addon config but I'm not immediately aware of how to enable the user to attach a file to their addon config, is that possible?

If we can come up with a set list of useful options we could use a sub-dict like this:

```json
"options": {
  "log_level": "info",
  "password": "",
  "update_lists_on_start": false,
  "admin_port": 4865,
  "dns_port": 53,
  "ssl": true,
  "certfile": "fullchain.pem",
  "keyfile": "privkey.pem",
  "interface": "",
  "ipv6": true,
  "ipv4_address": "",
  "ipv6_address": "",
  "virtual_host": "",
  "unbound_resolver": {
    "enabled": false,
    "verbosity": 0,
    "port": 5353,
    "do_ip4": true,
    "do_ip6": false,
    "do_tcp": true,
    "do_udp": true,
    "harden_glue": true,
    "harden_dnssec_stripped": true,
    "use_caps_for_id": false,
    "cache": {
      "min_ttl": 3600,
      "max_ttl": 86400
    },
    "prefetch": true,
    "num_threads": 1,
    "so_rcvbuf": "1m",
    "private_address": [
      "192.168.0.0/16",
      "169.254.0.0/16",
      "172.16.0.0/12",
      "10.0.0.0/8",
      "fd00::/8",
      "fe80::/10"
    ]
  },
  "hosts": []
}
```

Reply:
I agree with you on the totally over done addon config that would give.
Besides that still doesn't cover all most frequently used unbound usecases, for example it doesn't enable the DNS-TLS with forward option Unbound is frequently used for.

You could look into hooking into the /Shared directory and reading the config file from there?
Or maybe @frenck  has a beter idea...

I do however like adding unbound to add both recursive and forwarded-with-tls options 👍

Reply:
So whats the final word on the preferred method to expose more options? Would love to get this merged into the official Pi-Hole addon so I can stop running my own local 😄